app/lib/authentication/strategy.rb

```diff
@@ -12,12 +14,12 @@ def build_provider(provider_account)
         build_strategy(:provider_oauth2).new(provider_account, true) # TODO: make this configurable
       end
 
-      protected
-
       def build_strategy(type)
         inflected_type = Rails.autoloaders.main.inflector.camelize(type.to_s, {})
         strategy_class_name = "Authentication::Strategy::#{inflected_type}"
         strategy_class_name.constantize
+      rescue NameError
+        raise InvalidStrategyError
```

Comment:
In a perfect world, this would be done by returning a `Authentication::Strategy::Null` object that will always reject login.

Reply:
Also in another perfect world we don't use exceptions for flow control :nerd_face:

Reply:
In my head, this was error handling because not being able to find the auth strategy was leading to an error 500. So I added this to gracefully deal with the problem. However I see your point on this being flow control.

I created the `Authentication::Strategy::Null` class but I'm still using the exception to identify when the class doesn't exist. You probably still don't like it. Please suggest me a better way to do it.

https://github.com/3scale/porta/pull/4064/commits/10f90f72b3b91f2870fb963b468e36bd6d808027